File: 2026POTD/JANUARY/03-01-2026/1411.py
class Solution:
    def numOfWays(self, n: int) -> int:
        MOD = 10**9 + 7
        aba = 6
        abc = 6
        for _ in range(2, n + 1):
            newAba = (aba * 3 + abc * 2) % MOD
            newAbc = (aba * 2 + abc * 2) % MOD
            aba, abc = newAba, newAbc
        return (aba + abc) % MOD

File: 2026POTD/JANUARY/03-01-2026/1411.kt
class Solution {
    fun numOfWays(n: Int): Int {
        val MOD = 1000000007L
        var aba = 6L
        var abc = 6L
        for (i in 2..n) {
            val newAba = (aba * 3 + abc * 2) % MOD
            val newAbc = (aba * 2 + abc * 2) % MOD
            aba = newAba
            abc = newAbc
        }
        return ((aba + abc) % MOD).toInt()
    }
}